File: ABC069/C.java
import java.io.*;
import java.util.*;

class C{
    public void run() throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        br.readLine();
        String[] arg = br.readLine().split(" ");
        List<String> four = new ArrayList<>();
        List<String> two = new ArrayList<>();
        List<String> othe = new ArrayList<>();
        for(String a : arg){
            if(Integer.parseInt(a) % 4 == 0)
                four.add(a);
            else if(Integer.parseInt(a) % 2 == 0)
                two.add(a);
            else
                othe.add(a);
        }
        if(four.size() >= othe.size() - 1)
            System.out.println("Yes");
        else
            System.out.println("No");

    }
    public static void main(String... args) throws IOException{
        new C().run();
    }
}

File: ABC069/C.scala
object Main{
  def main(args:Array[String]) = {
    val tmp = scala.io.StdIn.readLine
    val list = scala.io.StdIn.readLine.split(" ").map(_.toInt).toList
    var four = List.empty[Int]
    var two = List.empty[Int]
    var oth = List.empty[Int]
    for(s <- list){
      if(s % 4 == 0)
        four = s :: four
      else if(s % 2 == 0)
        two = s :: two
      else
        oth = s :: oth
    }
    if(four.size >= oth.size - 1)
      println("Yes")
    else
      println("No")
  }
}
